ssm_tools/resolver.py

```diff
@@ -18,21 +21,84 @@ def __init__(self, args):
         # Construct boto3 session with MFA cache
         self.session = boto3.session.Session(profile_name=args.profile, region_name=args.region)
         self.session._session.get_component('credential_provider').get_provider('assume-role').cache = botocore.credentials.JSONFileCache(cli_cache)
+        
+        self.account_region = self.session.region_name
+        self.account_id = self.session.resource('iam').CurrentUser().user_id
```

Comment:
This should be `session.client('sts').get_caller_identity()['Account']` to make it work with Users as well as EC2 IAM Roles.